lib/Dialect/NDArray/Extensions/MeshShardingExtensions.cpp

```diff
@@ -689,6 +689,97 @@ struct ReshapeShardingInterface
     return {rank, utils::IteratorType::parallel};
   }
 };
+
+//===----------------------------------------------------------------------===//
+// PermuteDimsShardingInterface
+//===----------------------------------------------------------------------===//
+
+struct PermuteDimsShardingInterface
+    : public BaseShardingInterface<PermuteDimsShardingInterface,
+                                   imex::ndarray::PermuteDimsOp> {
+
+  FailureOr<ShardingOption>
+  getShardingOption(::mlir::Operation *op,
+                    ArrayRef<MeshSharding> operandShardings,
+                    ArrayRef<MeshSharding> resultShardings) const {
+    auto getShardingArray = [](ArrayRef<MeshSharding> shardings) {
+      ShardingArray axesArray(shardings[0].getSplitAxes().size());
+      for (auto [i, axes] : llvm::enumerate(shardings[0].getSplitAxes())) {
+        axesArray[i].append(axes.asArrayRef().begin(), axes.asArrayRef().end());
+      }
+      return axesArray;
+    };
+    if (!operandShardings.empty()) {
+      return ShardingOption(getShardingArray(operandShardings),
+                            operandShardings.front().getMeshAttr());
+    }
+    if (!resultShardings.empty()) {
+      return ShardingOption(getShardingArray(resultShardings),
+                            operandShardings.front().getMeshAttr());
+    }
+    return failure();
+  }
+
+  LogicalResult
+  addShardingAnnotations(::mlir::Operation *op, OpBuilder &b,
+                         const ShardingOption &shardingOption) const {
+    auto pdop = cast<PermuteDimsOp>(op);
+    auto srcShardOp = pdop.getSource().getDefiningOp<mesh::ShardOp>();
+    mlir::mesh::MeshSharding srcSharding;
+
+    if (srcShardOp) {
+      srcSharding = srcShardOp.getSharding();
+    } else {
```

Comment:
This check doesn't seem right, the source sharding must always follow the shardingOption, no matter if the sharding exists or not.